### PR TITLE
2960 template change warning

### LINKED ISSUE #2960: Temporarily: warn users about relationship deletion upon template change

### PR DESCRIPTION
**Pull request by fnocetti**
fixes #2960 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Comment by fnocetti** [comment]
The codeclimate issue is related to code duplicated in the entity and document form from time ago.
Maybe at some point in the future it will make sense to deduplicate that code?

**Comment by fnocetti** [comment]
Good catch @mfacar !! Thank you!

**Comment by daneryl** [comment]
@konzz @fnocetti did not see that konz self assigned, i was revieweing and found that if you change the template the warning shows properly but if you change it back without saving the warning is still there, i think this should not be the case. i let konz do the rest of the review.

**Comment by fnocetti** [comment]
@daneryl you're right, it was somewhat on purpose. The implementation is based on the field being dirty, that doesn't care about it's actual value but the user interaction.
I couldn't find a quick good way to take in count what was the original value because the form doesn't keep that information.
If you can give me a hand with that, I'll greatly appreciate it!! :D

File: app/react/Documents/components/DocumentSidePanel.js [hunk 1]
```diff
@@ -267,10 +268,20 @@ export class DocumentSidePanel extends Component {
...
+                  );
                 }
                 if (docBeingEdited && isEntity) {
-                  return <EntityForm storeKey={this.props.storeKey} />;
+                  return (
+                    <EntityForm
+                      storeKey={this.props.storeKey}
+                      initialTemplateId={this.initialTemplateId}
+                    />
+                  );
```
[the remark this answers is not in the document]
Comment (fnocetti) [reply app/react/Documents/components/DocumentSidePanel.js:284]: We saw this duplication when working in this PR and agree that we need to resolve it but after checking we think the change might be a bit bigger than we expect and delay this PR unnecessary. Given that, I'm going to report a Tech Debt issue to work on it.

File: app/react/Metadata/components/specs/__snapshots__/MetadataForm.spec.js.snap [hunk 2]
```diff
@@ -60,6 +60,22 @@ exports[`MetadataForm should display custom title label 1`] = `
...
+              icon="exclamation-triangle"
+            />
+            <Connect(Translate)
+              context="System"
+            >
+              Changing the type will erase all connections to this entity.
+            </Connect(Translate)>
+          </span>
+        </ShowIf>
+      </li>
```
[the remark this answers is not in the document]
Comment (fnocetti) [reply app/react/Metadata/components/specs/__snapshots__/MetadataForm.spec.js.snap]: @mfacar just implemented this change.

File: app/react/Metadata/components/specs/MetadataForm.spec.js [hunk 3]
```diff
@@ -113,12 +113,40 @@ describe('MetadataForm', () => {
   });
 
   describe('on template change', () => {
+    function assessWarningVisibility(visible) {
+      const warning = component.find({
+        children: 'Changing the type will erase all connections to this entity.',
+      });
+      expect(warning.parents().get(1).props.if).toBe(visible);
+    }
+
```
[the remark this answers is not in the document]
Comment (fnocetti) [reply app/react/Metadata/components/specs/MetadataForm.spec.js]: @mfacar just implemented this change.

File: app/react/Metadata/components/specs/__snapshots__/MetadataForm.spec.js.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (fnocetti) [reply app/react/Metadata/components/specs/__snapshots__/MetadataForm.spec.js.snap]: We'll take a look

**Pull request merged**